programs/bpf_loader/src/lib.rs

```diff
@@ -273,55 +269,67 @@ fn process_instruction_common(
     use_jit: bool,
 ) -> Result<(), InstructionError> {
     let log_collector = invoke_context.get_log_collector();
-    let program_id = invoke_context.transaction_context.get_program_key()?;
-
-    let keyed_accounts = invoke_context.get_keyed_accounts()?;
-    let first_account = keyed_account_at_index(keyed_accounts, first_instruction_account)?;
-    let second_account = keyed_account_at_index(keyed_accounts, first_instruction_account + 1);
-    let (program, next_first_instruction_account) = if first_account.unsigned_key() == program_id {
-        (first_account, first_instruction_account)
-    } else if second_account
-        .as_ref()
-        .map(|keyed_account| keyed_account.unsigned_key() == program_id)
+    let transaction_context = &invoke_context.transaction_context;
+    let instruction_context = transaction_context.get_current_instruction_context()?;
+    let program_id = transaction_context.get_program_key()?;
```

Comment:
Not directly related to this PR but calling `get_program_key()` on a transaction context seems a little strange since there could be may program keys in a transaction.  Would be more clear if `get_program_key()` took an `InstructionContext` or was called `get_current_program_key()`.  Same goes for `get_loader_key()`

Reply:
#23191